### PR TITLE
#162163270 Implement threaded comments

### PR DESCRIPTION
**Pull request by Kasulejoseph**
**What does this PR do?**
This pull request enables users to comment on authors' articles

**Description of tasks to be completed?**
- write tests for the feature
- A user can only delete or update his comment 
- Logged in user can view a list of all comments
- Have these endpoints working:
```
POST/GET /api/articles/:slug/comments/ --> create or get comments
POST/GET /api/articles/:slug/comments/:id/ --> create or get replies
PUT/DELETE /api/articles/:slug/comments/:id/ --> edit or delete an comment
```
**How should this be tested manually?**
- Install the requirements with pip install -r requirements.txt
- To run tests, run ./manage.py test
- Run python manage.py runserver

*Test the following endpoints in postman*
1. Create a comment on article --> POST /api/articles/:slug/comments/ 
`comment body`
```
{
    "comment": {
"comment_body": "Hey, nooo"	
}
} 
```
`output`
screen shot 2018-12-15 at 18 35 31
2. get all comments  --> GET /api/articles/:slug/comments/ 
screen shot 2018-12-15 at 18 39 05

3. reply to a comment --> POST /api/articles/:slug/comments/ : id>/
`reply body`
```
{
    "reply": {
"comment_body": "Hey, bro"	
}
}
```
`output`
screen shot 2018-12-15 at 18 45 50

4. get replies --> GET /api/articles/:slug/comments/ : id>/
screen shot 2018-12-15 at 18 50 46

5. user can only edit his comment
screen shot 2018-12-15 at 18 54 40

6. test coverage
screen shot 2018-12-15 at 18 55 30

**Relevant pivotal tracker stories**
#162163270

### CODE REVIEW

**Review by kafuuma — APPROVED** [review]
I tested this
Multithreading works well

**Review by bibangamba — APPROVED** [review]
great work @Kasulejoseph

**Review by KicaRonaldOkello — APPROVED** [review]

File: authors/apps/articles/exceptions.py [hunk 1]
```diff
@@ -1,3 +1,14 @@
...
 """
+from rest_framework.exceptions import APIException
+
+class PermisionDenied(APIException):
+    status_code = 403
+    default_detail = 'You dont have permission to perform this action'
+
+class CommentDoesNotExist(APIException):
+    status_code = 404
+    default_detail = "Failed, comment or article doesnot exist"
```
Comment (collin5) [inline authors/apps/articles/exceptions.py]: Typo, should be "... does not exist"

**Review by collin5 — CHANGES_REQUESTED** [review]

**Review by collin5 — APPROVED** [review]

File: authors/apps/articles/exceptions.py (hunk 1 above)
Comment (Kasulejoseph) [reply authors/apps/articles/exceptions.py]: I have implemented the feedback

**Pull request merged**